drivers/timer/nrf_rtc_timer.c

```diff
@@ -15,14 +16,26 @@
 
 #define RTC NRF_RTC1
 
-#define COUNTER_MAX 0x00ffffff
+/*
+ * Compare values must be set to at least 2 greater than the current
+ * counter value to ensure that the compare fires.  Compare values are
+ * generally determined by reading the counter, then performing some
+ * calculations to convert a relative delay to an absolute delay.
+ * Assume that the counter will not increment more than twice during
+ * these calculations, allowing for a final check that can replace a
+ * too-low compare with a value that will guarantee fire.
+ */
+#define MIN_DELAY 4
```

Comment:
The RTC IRQ could be pre-empted by Bluetooth and zero-latency IRQs.
IMHO we should an algorithmic solution here.

Reply:
No, it can't.  Both the original spin_lock and the irq_lock() block the Bluetooth IRQs, which is why this is a problem.

Reply:
Ok. I missed the irq_lock().